### PR TITLE
Add repository root README and CONTRIBUTING markdown templates

### PR DESCRIPTION
**Pull request by QuintinWillison**
Working under [this internal ticket](https://ably.atlassian.net/browse/SDK-700).

I've added a lot of reviewers here as I know there are a lot of people across the organisation who will have an interest in what we conform across our repositories. Please feel free to ignore this pull request if you don't have time or would prefer not to contribute. Thanks!

In the back of my mind is the idea that we _might_ consider adding workflows to client library SDK repositories that check if non-optional headings exist and are in the order specified here. Of limited benefit once they're all conformed but might be interesting. 🤔

### CODE REVIEW

**Review by tomkirbygreen — APPROVED** [review]
Looks good to me.

**Review by marklewin — COMMENTED** [review]
I think this is a great start @QuintinWillison. My own experience of watching developers look at SDK README's though is that they want to see code first and foremost before they'll even consider reading anything else, so I'd suggest the following general outline:

```
## Overview

One or two sentences describing what the SDK does, link to more complete overview on ably.com

## Getting Started

### Prerequisites
 List here and link to "Supported Platforms" later in the page

### Installation 	

### Usage examples
 Code examples showing common use cases and link to demo app (if there is one)

## Find out more

### Documentation

### Other resources

## Supported platforms

## Known limitations

## Getting help

## Contributing to this project

## Credits
```

**Review by niksilver — APPROVED** [review]
LGTM.

**Review by AndyNicks — APPROVED** [review]
LGTM

**Review by ben-xD — COMMENTED** [review]
Would it be a good idea to limit the usage of these templates to a few repos for a few months, to measure how beneficial (or costly) standardisation can be? `README`s may also become less relevant if we move over to a developer friendly documentation (making these consistent between each other is also arguable). Some repos link to their documentation immediately, and have a very minimal README. Also, I could not find any one in open source attempting to make their documentation consistent between different language SDKs. What are the benefits/ costs of this? I believe this sprung up a slack message from @mattheworiordan

The structure of the template is highly debatable. My opinion is it introduces friction/ work but no benefit (happy to be proven wrong). **A checklist of useful things or list of considerations a readme might want to contain would help.**

File: templates/sdk-readme.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Not all repositories will have this, but if there is an example app then this is often a good entry point for developers wishing to gain familiarity before attempting to integrate it into their own application codebase.
+
+Example:
+[ably-flutter](https://github.com/ably/ably-flutter#running-the-example)
+
+## Installation
+
+How to install this SDK into an application project (whether a new or existing codebase), covering all supported package management and distribution systems.
+
+## Usage Examples
```
[the remark this answers is not in the document]
Comment (QuintinWillison) [reply templates/sdk-readme.md:58]: An interesting proposal, but perhaps counter to what @marklewin suggests in [his review](https://github.com/ably/ably-common/pull/88#pullrequestreview-693843000) when he says:

> My own experience of watching developers look at SDK README's though is that they want to see code first and foremost before they'll even consider reading anything else

So, clearly, tastes vary on this. 🤷

File: templates/sdk-readme.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+This should be formatted as a bulleted list and each feature should link to an issue in this repository under which the work will be done when we come to implement it. This allows people to comment on or watch the issue if they are interested.
+
+Example:
+[ably-flutter](https://github.com/ably/ably-flutter#known-limitations)
+
+## Requirements
+
+What is required to build a project using this SDK. Typically a list of prerequisites.
+
+May also include runtime requirements specific to particular platforms, expanding on what's detailed at a higher level under [Supported Platforms](#supported-platforms).
```
[the remark this answers is not in the document]
Comment (QuintinWillison) [reply templates/sdk-readme.md:45]: I, indeed, deliberated over that too. Formally combining these two is definitely something we should consider. I would be interested to hear a take from others on this.

**Review by tomczoink — APPROVED** [review]
Looks good! 😄

**Comment by thisisjofrank** [comment]
Agree with @marklewin on the ordering, I usually want getting started/prerequisites/installation right up top

**Comment by QuintinWillison** [comment]
Thanks, @tomczoink and @thisisjofrank 😁 

@marklewin, @ben-xD and myself just got of a call and I've pushed a couple of commits as a result. I'll re-request review from those who had approved already.

**Review by ben-xD — APPROVED** [review]

**Review by marklewin — APPROVED** [review]
LGTM!

**Review by tbedford — APPROVED** [review]

File: templates/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Ably Templates
+
+| Template | Target | Target Name | Synopsis |
```
Comment (tbedford) [inline templates/README.md:3]: Minor suggestion. Would the following titles be simpler?

Template | Repository | Filename | Description

Comment (QuintinWillison) [reply templates/README.md:3]: I like it. I'll push a commit. Thanks.

Comment (QuintinWillison) [reply templates/README.md]: Done: https://github.com/ably/ably-common/pull/88/commits/ad0cbdab7d97dd8cb8e0f4badc92fe3efbc8b9bb

File: templates/sdk-readme.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (QuintinWillison) [reply templates/sdk-readme.md]: Done: https://github.com/ably/ably-common/pull/88/commits/0ccc91698c48dc1018a4c88766cc7400fa0d46d0

**Comment by QuintinWillison** [comment]
Thanks all for your input on this pull request. I think it's a good step forward, setting some good foundations. That's not to say it needs to stay static - we should continue to evolve this as needs arise and questions need answering.

**Pull request merged**